### ISSUE TITLE
Difference in implementation and paper

### ISSUE THREAD

**Issue by germayneng** [microsoft/StemGNN#12]
Good day,

Thank you for the paper. I noticed that there are some difference in implementation described in the paper as well as the source code, Is this intentional? For example, the spectral sequential cell defined in the paper uses a convo 1D operation but i cant seem to see it in the code (unless i am missing something ) 

Will be good to explain! thanks!

**Comment by WooheonHong** [comment]
I totally agree with you..
It's nothing like a paper.

**Comment by lsrock1** [comment]
I agree, is there any reason for these differences?

**Comment by lsrock1** [comment]
I think the authors use Chebyshev Polynomials to reduce time complexity (laplacian(n^3)).
(review 3 of neurips)
ref [https://proceedings.neurips.cc/paper/2020/file/cdf6581cb7aca4b7e19ef136c6e601a5-Review.html](https://proceedings.neurips.cc/paper/2020/file/cdf6581cb7aca4b7e19ef136c6e601a5-Review.html)
ref [https://towardsdatascience.com/an-introduction-to-graph-neural-networks-e23dc7bdfba5](https://towardsdatascience.com/an-introduction-to-graph-neural-networks-e23dc7bdfba5)

**Comment by WooheonHong** [comment]
Output layers using GLU and fully connected layers are also not implemented... 
Is there anyone who reproduced the paper's performance? 
I can't reproduce metr-la and ECG data performance in any way.

**Comment by germayneng** [comment]
> Is there anyone who reproduced the paper's performance?

I think the current pytorch implementation is not following the paper's defined architecture. Perhaps we can try to replicate the exact architecture to see if the results will be closer?

**Comment by lsrock1** [comment]
I tried but this paper lacks detail about implementation.
The overall concept is clear but the core detail is dismissed.

**Comment by razvanc92** [comment]
I've also tried to reproduce the results on PEMS 3,4 and 7 but without success.

**Comment by Einsturing** [comment]
Hi, I'm reading the paper and code rencently and also find the difference like the loss F you've described.Have you found a solution to the problem?

**Comment by lixus7** [comment]
why not do some ablation experiment about single Y1 and single Y2?

**Comment by wangzhengzhengkai** [comment]
> > > 看来作者在他们执行的任务上犯了一些错误。比如pems07是一个交通速度数据，作者把它当成一个交通流任务，同时pems04实验的结果肯定是一个交通流任务，但是作者把它作为一个交通速度任务来运行。
> > 
> > 
> > @superarthurlx我认为这些概念越来越模糊，可能是因为它们与模型相同。例如，STGNN[1] 也使用 METR-LA 和 PEMS-BAY 作为交通流预测数据集，但是它们记录交通速度。
> > [1] 王 X，马 Y，王 Y，等。基于时空图神经网络的交通流预测[C]//Proceedings of The Web Conference 2020. 2020: 1082-1​​092.
> 
> 检查 table2 或 pems07.csv 文件你可以找到不同的。流量数据应该是整数，而速度数据是浮点数，通常小于 70，这是 pem 区域的速度限制。
> 
> 这是我的一些复制品，仍然存在许多差异，例如 mape 值。==！ ![图片](https://user-images.githubusercontent.com/40826115/139094506-59abddde-52e2-4faf-a823-19cd4d7f70e0.png)
Hello, can you tell me the parameters you are replicating on the ECG dataset? I can't get the results you show.

**Comment by semink** [comment]
Hello,

So, no one has succeeded in reproducing the results on their paper?

**Comment by stemill** [comment]
Also interested in whether or not anyone has successfully reproduced output? Sounds like there are some outstanding discrepancies between the paper and the code?

**Comment by zezhishao** [comment]
I found that the calculation of metrics in StemGNN seems to be unfairly and inconsistent with baselines, such as the calculation in [GWNet](https://github.com/nnzhan/Graph-WaveNet/blob/6b162e80c59a1d494809252eca055cff93dc66b1/train.py#L156).

**Comment by onceCWJ** [comment]
I cannot reproduce the same results as in their paper, too.

**Comment by onceCWJ** [comment]
I cannot reproduce the same results as in their paper, either.
